### ISSUE TITLE
Only on/off configurable option can be configured to set Plesk add-ons

### ISSUE THREAD

**Issue by ioxak** [vbaranovskiy-plesk/whmcs-plugin#33]
It seems that only on/off configurable options in WHMCS (tick selection) can be configured to set Plesk add-ons using the "Plesk <Panel add-on name>".

This seems to be in contradiction to document that is provided as follows:
https://docs.whmcs.com/Plesk

Please see especially the part "Panel Addons" and the following image:
https://docs.whmcs.com/File:Plesk_config_options.png

However this Plesk WHMCS-plugin doesn't seem to work as shown in the image, as only working "option type" is "on/off" and it seems that only that selected you can enter "Plesk <Panel add-on name>" (i.e. "Plesk 1 GB extra space") to option field.

This results very long and confusing lists of configurable options both in WHMCS order pages as well as on invoices as there can be long lists of configurable options as follows:

Web hosting package 1 GB
-1 GB extra disk space: no
-2 GB extra disk space: yes +1 EUR/mo
-3 GB extra disk space: no
-4 GB extra disk space: no
-10 GB extra bandwidth: no
-20 GB extra bandwidth: no
-30 GB extra bandwidth: yes +3 EUR/mo
-40 GB extra bandwidth: no

This module should work also with WHMCS configurable option types "dropdown" and "radio" where you could set module names in options just like shown in the document picture:
https://docs.whmcs.com/File:Plesk_config_options.png

There could be very same logic but instead of putting Plesk <Panel add-on name> on "option name" field it should be put into one or several of the options like this:

Option name: Additional disk spaces
Option 1: no
Option 2: Plesk 1 GB disk space
Option 2: Plesk 2 GB disk space
Option 3: Plesk 3 GB disk space

where each of the "X disk space" would match the corresponding add-on in Plesk. This would allow maximizing the options in WHMCS order forms using i.e dropdown menus and minimize the required space for example in order forms, shopping cart and invoices like as follows:

Web hosting package 1 GB
- Extra disk space: 2 GB extra disk space
- Extra bandwidth: no

Could you please fix the WHMCS-plugin so that it would work also with this kind of logic using drop down menus and radio buttons as well? This is a very essential feature that is required for most of the web hosting service providers using WHMCS. Thank you!

**Comment by WHMCSChance** [comment]
Hey @ioxak ,

I wanted to reach out to you and let you know that I was able to get the dropdown type config options working with the Plesk Addons in WHMCS.  I would recommend reaching out to our Tech Support team to further assist you with getting some assistance and a better understanding on how to get this setup and working.

Thanks!

https://www.whmcs.com/submit-a-ticket/

**Comment by ioxak** [comment]
Thank you.

I already did try to solve this problem trough WHMCS support, but their response at the time was that, because WHMCS Plesk module is no longer developed by WHMCS but instead Plesk, they cannot help me with this matter. So this is the reason I write here in the first place, but it seems that official Plesk support isn't having much of an activity towards their WHMCS module.

But could you please share your information how to get dropdown type config options working with Plesk add-ons in case you have been managed to get it to work somehow? Do you need any extra scripts or modifications to the module, or is it just formulating the drop down menu option names correctly? (And if, how?)

**Comment by WHMCSChance** [comment]
Hey @ioxak ,

Unfortunately I do not feel this would be the most appropriate venue to provide technical support through.  Please open a support ticket with our tech support and provide me with the TicketID here and I will follow up with you.

Thanks!

**Comment by ioxak** [comment]
Hi Chance,

I already have a ticket submitted, but it was closed unresolved (considering for this specific issue) because of the mentioned reason). Can you possibly recheck and reopen the ticket [Ticket ID: QEM-093416] or should I still open a new one?

**Comment by WHMCSChance** [comment]
Sorry for the delay, I have your ticket and I will respond to it shortly.

Thanks!

**Comment by chrisduk112** [comment]
HI

Can someone post on here how to get config options to work with Plesk?

We use Windows Server with Plesk Onyx

We created a service plan in Plesk called Custom and set diskspace to 1mb and unlimited bandwidth 0 databases, 0 email accounts etc.

I then created a hosting plan in WHMCS and assigned it to "custom" along with custom fields that match addons, for example

In Plesk we have addon "1 GB disk space (add-on)"  thats set to +1GB
we did this for 1GB/2GB/3GB addons

in WHMCS we create a config field called "diskspace"  
options are:

Plesk 1 GB disk space|1GB Space
Plesk 2 GB disk space|2GB Space
Plesk 3 GB disk space|3GB Space

however when clicking change plan or create it doesn't create the addons so the hosting plan in Plesk remains 1MB space.

How do we fix this?
